### PR TITLE
feat: Schedule Return Dish Notifications: #125

### PR DESCRIPTION
**Pull request by MSNanda515**
**Description**
PR for issue #125 
Schedule notifications for returning dishes

**Required Items**

- [x] Title matches [../CONTRIBUTING.md](contributing guidelines)
- [ ] Linked Issue: #XYZ
- [ ] Code Reviewed
- [x] Manually tested
- Style
  - [ ] No Lint Errors
  - [ ] Quality Comments
- Tests
  - [ ] Unit Tests Added
  - [ ] CI/CD passes
- UI
  - [ ] Picture of before/after UI (if applicable)

### CODE REVIEW

**Comment by MSNanda515** [comment]
@MacMalainey @Afraz0Khan 
Made the following changes to the schema:
- Added the email field to the user documents
- Added a new collection called config. It contains a document "admin" that further contains the field required such as "emailNotificationDelay"

**Comment by MSNanda515** [comment]
Removed all the secrets that were originally included.
Also, We will need an email account to complete the email notification functionality.

**Comment by Afraz0Khan** [comment]
Looks good. I'll contact Mac to check with the email placeholder and see what we can do.

**Comment by MacMalainey** [comment]
Please fix merge errors

> Looks good. I'll contact Mac to check with the email placeholder and see what we can do.

The user's email should be accessible using the Firebase Admin SDK, since transactions point to the UUID for the user. Please use that instead of a placeholder.

Unless I am misunderstanding what the placeholder email is for

**Comment by MSNanda515** [comment]
> Please fix merge errors
> 
> > Looks good. I'll contact Mac to check with the email placeholder and see what we can do.
> 
> The user's email should be accessible using the Firebase Admin SDK, since transactions point to the UUID for the user. Please use that instead of a placeholder.
> 
> Unless I am misunderstanding what the placeholder email is for

When you say UUID for the user, do you mean the document id will be the same as the CCID for the users, or do you mean that the user model will have a field containing the email (right now, there is no email field in the user's schema).

**Comment by MSNanda515** [comment]
Added code to get the user's model from Firebase Auth instead of Firestore. I also removed the email entry (I added before) from the User's schema in the software design document.

File: server/controllers/scheduledJobsController.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import firestore from "../db.js";
+import emailService from "../services/emailService.js";
+
+import transactionService from "../services/transactionService.js";
+import userService from "../services/userService.js";
+
+const scheduledJobsController = {
+
+    // Schedules a job to send the reminder to return the borrowed dish
+    dishCheckoutReturnReminder: async function (transId) {
```
Comment (MacMalainey) [inline server/controllers/scheduledJobsController.js:12]: Sorry, would we be able to do this as a batch job?

I.e. every Friday at 2:00pm check for transactions that are minimum N-days old and send an email to the users for them all at once (that way we can batch multiple missing dishes into one email together)

File: server/routes/scheduledJobs.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Router } from 'express'
+import scheduledJobsController from '../controllers/scheduledJobsController.js';
+
+const router = Router()
+
+// Schedules the reminder to return the dish
+// Requires transId, userId, dishId as query params
+router.get('/dishCheckout', async (req, res, next) => {
```
Comment (MacMalainey) [inline server/routes/scheduledJobs.js:8]: We shouldn't be triggering the reminder, we should be running a job every N-days to run a query on all outdated transactions

File: server/services/emailService.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import nodemailer from "nodemailer"
```
Comment (MacMalainey) [inline server/services/emailService.js:1]: Should use the firebase email service

Comment (MacMalainey) [reply server/services/emailService.js:1]: Will also reach out to DishZero regarding aquiring a more official email, but that can occur later

File: server/controllers/scheduledJobsController.js (hunk 1 above)
Comment (MSNanda515) [reply server/controllers/scheduledJobsController.js]: Updated it to a batch job. The config needs to be in cron syntax for it to be valid.

File: server/services/emailService.js (hunk 3 above)
Comment (MSNanda515) [reply server/services/emailService.js]: Removed nodemailer. Will use the firebase email service once obtain details regarding the sender's email address.

**Review by Afraz0Khan — APPROVED** [review]

**Pull request merged**